### PR TITLE
Update rust-bitcoin to 0.20

### PR DESCRIPTION
**Pull request by jb55**
This includes the new Amount apis, etc

I'm merging this into gdk_integration because it has some stuff that still needs to be upstreamed (rust-jsonrpc, rust-bitcoincore-rpc

Signed-off-by: William Casarin <jb55@jb55.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,17 +9,13 @@ edition = "2018"
...
 stderr_logger = ["stderrlog"]
 
 [dependencies]
-bitcoin = "0.18"
-bitcoincore-rpc = "0.7.0"
-bitcoin_hashes = { version = "0.3", features = [ "serde" ] }
-secp256k1 = { version = "0.12", features = [ "rand" ] }
-jsonrpc = "0.11"
+bitcoin = { version = "0.20", features = [ "use-serde" ] }
+bitcoin_hashes = { version = "0.7", features = [ "serde" ] }
```
Comment (stevenroose) [inline Cargo.toml:18]: You can remove this one. It's accessible as `use bitcoin::hashes::{...};` instead. So you make sure the versions can't get out of sync.

File: Cargo.toml [hunk 2]
```diff
@@ -36,15 +32,18 @@ android_log = { version = "0.1", optional = true }
 url = "1.7.2"
 backtrace = "0.3"
 
+[dependencies.jsonrpc]
+git = "https://github.com/jb55/rust-jsonrpc"
+rev = "5e4dd4817641aa2f5b9981b4ea7f3bca52e0b3d9"
+
+[dependencies.bitcoincore-rpc]
+git = "https://github.com/jb55/rust-bitcoincore-rpc"
+rev = "f9eea91f82b3812d0c82810f74edba4259ec6939"
```
Comment (stevenroose) [inline Cargo.toml:41]: Please upstream if you have meaningful patches!

File: Cargo.toml [hunk 3]
```diff
@@ -36,15 +32,18 @@ android_log = { version = "0.1", optional = true }
...
+
 [dependencies.elements]
 version = "0.7.1"
 optional = true
 
-[dependencies.liquid-rpc]
-git = "https://github.com/stevenroose/rust-liquid-rpc.git"
-branch = "master"
-optional = true
```
Comment (stevenroose) [inline Cargo.toml:46]: Are you dropping Liquid support?

**Review by stevenroose — COMMENTED** [review]
Why are you removing the liquid-rpc crate?

Also, do you plan to PR the jsonrpc and bitcoincore-rpc changes?

File: Cargo.toml (hunk 1 above)
Comment (jb55) [reply Cargo.toml:18]: good call, I think I remember having issues with this but I could try again...

File: Cargo.toml (hunk 3 above)
Comment (jb55) [reply Cargo.toml:46]: ah no it there were many elements serialization issues when I upgraded rust-bitcoin and I didn't get around to fixing it. I have this on my TODO

**Pull request closed without merging**